pypesto/visualize/misc.py

```diff
@@ -80,18 +84,19 @@ def process_result_list(
                 legends.append('Result ' + str(i_leg))
         else:
             # legends were passed by user: check length
-            if isinstance(legends, list):
+            try:
+                if isinstance(legends, str):
```

Comment:
if this is true it already means that 'List of results passed and list of labels do not have the same length.', right? As it's a multiple result case + only a string was passed as a legend

Reply:
The string could have matching length, e.g. "ABC" for 3 results. 

The idea was that a string of length != len(results) would throw an error, but a string with the appropriate number of characters would be used for the legend. We could also raise an error for any string, but this would not be consistent with the type hint for the legend argument of visualize.waterfall: legends: Optional[Union[Sequence[str], str]] = None, because a string is a Sequence.